### PR TITLE
chore: upgrade the version of COG

### PR DESCRIPTION
**Pull request by Bento007**
## Reason for Change

- #901 

## Changes

- update the version of COG for schema 5.1
- install the cli from the repo. This is to make sure we don't break the installation process when making changes.
## Testing
- verify unit tests pass
-

### CODE REVIEW

**Review by nayib-jose-gloria — COMMENTED** [review]
getting the following error trying to install locally 

```
Processing /Users/ngloria/code/single-cell-curation/cellxgene_schema_cli
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [1 lines of output]
      error in cellxgene-schema setup command: 'install_requires' must be a string or list of strings containing valid project/version requirement specifiers; Parse error at "'-r ./req'": Expected W:(abcd...)
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.


```

**Review by nayib-jose-gloria — CHANGES_REQUESTED** [review]
It seems like install_requires in setup.py does not support using `-r <path>` to install packages from other requirement files when building the package. 

We'll need to either 1) create a function in setup.py to parse the requirements file for nested requirements or 2) revert this change

I don't think it's too bad to have two spots to update COG, but it probably wouldn't take too long to implement option 1.

**Review by nayib-jose-gloria — APPROVED** [review]

**Comment by Bento007** [comment]
> getting the following error trying to install locally
> 
> ```
> Processing /Users/ngloria/code/single-cell-curation/cellxgene_schema_cli
>   Preparing metadata (setup.py) ... error
>   error: subprocess-exited-with-error
>   
>   × python setup.py egg_info did not run successfully.
>   │ exit code: 1
>   ╰─> [1 lines of output]
>       error in cellxgene-schema setup command: 'install_requires' must be a string or list of strings containing valid project/version requirement specifiers; Parse error at "'-r ./req'": Expected W:(abcd...)
>       [end of output]
>   
>   note: This error originates from a subprocess, and is likely not a problem with pip.
> error: metadata-generation-failed
> 
> × Encountered error while generating package metadata.
> ╰─> See above for output.
> ```

Push tests will now test for this by installing the cli from the commit.

File: .github/workflows/push_tests.yml [hunk 1]
```diff
@@ -49,8 +49,8 @@ jobs:
             ${{ runner.os }}-pip-
       - name: Install dependencies
         run: |
-          pip install -r ./cellxgene_schema_cli/requirements.txt
           pip install -r ./requirements-dev.txt
+          pip install -e ./cellxgene_schema_cli
```
Comment (Bento007) [inline .github/workflows/push_tests.yml:53]: @nayib-jose-gloria I made this change post your approval.

**Pull request merged**